### PR TITLE
Adding support for dynamic settings from Consul.

### PR DESCRIPTION
**Pull request by hozn**
I added a new dynamic settings backend to support Consul (https://www.consul.io/).  We wanted a configuration project that would support dynamic config and config files and environment vars; this looked perfect except we are using Consul instead of Redis for configuration.  Hopefully this is something that can be merged in.  I included a test based very closely on the redis tests and updated docs.  (I made some other grammatical changes in docs too.)

Thanks!

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ def read(fname):
 
 YAML_REQUIRES = ['PyYAML==3.11']
 REDIS_REQUIRES = ['redis==2.10.5', 'six==1.10.0']
+CONSUL_REQUIRES = ['consulate==0.6.0']
 ALL_REQUIRES = YAML_REQUIRES + REDIS_REQUIRES
```
Comment (drgarcia1986) [inline setup.py]: add `CONSUL_REQUIRES` in `ALL_REQUIRES`
```python
ALL_REQUIRES = YAML_REQUIRES + REDIS_REQUIRES + CONSUL_REQUIRES
```

File: simple_settings/dynamic_settings/consul_reader.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    """
+    _default_conf = {
+        'host': consulate.DEFAULT_HOST,
+        'port': consulate.DEFAULT_PORT,
+        'scheme': consulate.DEFAULT_SCHEME
+    }
+
+    def __init__(self, conf):
+        self._default_conf.update(conf)
+        self.session = consulate.Consul(host=self._default_conf['host'], port=self._default_conf['port'],
```
Comment (drgarcia1986) [inline simple_settings/dynamic_settings/consul_reader.py]: I preffer this style
```python
self.session = consulate.Consul(
    host=self._default_conf['host'], 
    port=self._default_conf['port'],
    datacenter=self._default_conf.get('datacenter'),
    token=self._default_conf.get('token'), 
    scheme=self._default_conf['scheme']
)
```

File: simple_settings/dynamic_settings/consul_reader.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+                                        datacenter=self._default_conf.get('datacenter'),
+                                        token=self._default_conf.get('token'), scheme=self._default_conf['scheme'])
+
+    def get(self, key):
+        try:
+            result = self.session.kv[key]
+        except KeyError:
+            result = None
+        return result
```
Comment (drgarcia1986) [inline simple_settings/dynamic_settings/consul_reader.py]: simplify 😄 
```python
def get(self, key):
    try:
        return self.session.kv[key]
    except KeyError:
        return
```

**Comment by drgarcia1986** [comment]
Awesome 👏 
You can fix the errors that are appearing in travis?

About "dynamic" settings, I working in some features about this, some like `pattern` to read only settings with this pattern in dynamic reader and (maybe) methods to write in dynamic reader (maybe).
But, if you need to use the consul reader I can generate a new version and update pypi.

File: setup.py (hunk 1 above)
Comment (hozn) [inline setup.py]: Ok, done.

File: simple_settings/dynamic_settings/consul_reader.py (hunk 2 above)
Comment (drgarcia1986) [inline simple_settings/dynamic_settings/consul_reader.py]: and this fix flake8 issues ☝️

Comment (hozn) [inline simple_settings/dynamic_settings/consul_reader.py]: Sure, that's fine.  Updated.

File: simple_settings/dynamic_settings/consul_reader.py (hunk 3 above)
Comment (hozn) [inline simple_settings/dynamic_settings/consul_reader.py]: Sure, done.  (I used "return None" to be slightly more explicit.)

**Comment by hozn** [comment]
I think last pushes should fix the flake8 problems.

**Comment by drgarcia1986** [comment]
@hozn hum, I think the `consul` test don't working in travis because have a "consul server" dependency, any chance to run one in travis environment ?

**Comment by hozn** [comment]
Sorry, I really don't know how to do that; I've never used travis ...  I'm sure it's trivial to setup a docker container; that's how I'm using it in testing locally.  If you want to provide guidance/example for redis, I can help.

**Comment by hozn** [comment]
For reference, I am using https://hub.docker.com/r/progrium/consul/

And running a simple consul container is:
```
docker run -p 8400:8400 -p 8500:8500 -p 8600:53/udp -h node1 progrium/consul -server -bootstrap
```

**Comment by drgarcia1986** [comment]
for `redis` is easy, I use [travis services](https://docs.travis-ci.com/user/database-setup/)

I'm looking for a solution to this problem, you can wait one day (maybe two) ?

**Comment by hozn** [comment]
I can wait, sure.  We are just using a locally forked version with this patch applied in interim anyway.

**Comment by drgarcia1986** [comment]
@hozn done in #33 .
Please rebase your branch with master.

**Comment by hozn** [comment]
Thank you! (Great project, by the way!)

**Comment by drgarcia1986** [comment]
Many Thanks @hozn 😄 
After you rebase your branch and the build pass with success I will merge this PR

**Comment by hozn** [comment]
Ok, I did a forced push with the rebased branch; hope that'll work / not break things (as opposed to a merge, which is admittedly where I have more experience).

**Comment by hozn** [comment]
Interesting; I'm confused as to why tests failed for one of the builders (most passed, but 2.7 is still pending).

**Comment by drgarcia1986** [comment]
Hum, fail in `pypy3`, `consulate` is compatible with pypy3?

**Comment by drgarcia1986** [comment]
(if not, I drop support to pypy3 in travis build, no problem)

**Comment by hozn** [comment]
Yeah, perhaps it is not compatible! Thanks for fixing.

**Comment by drgarcia1986** [comment]
Ok, thanks again @hozn 👍

**Pull request merged**